Extension/src/common.ts

```diff
@@ -728,29 +730,27 @@ export function isExecutable(file: string): Promise<boolean> {
 }
 
 export function allowExecution(file: string): Promise<void> {
-    return new Promise<void>((resolve, reject) => {
+    return new Promise<void>(async (resolve, reject) => {
```

Comment:
Is it valid to pass an async lambda to a Promise?  Shouldn't this Promise be removed and allowExecution be converted to async?